source/dotnet/Services/Application/IIntegrationEventPublisher.cs

```diff
@@ -21,5 +21,7 @@ public interface IIntegrationEventPublisher
 {
     Task PublishAsync(ProcessCompletedEventDto processCompletedEvent);
 
-    Task PublishAsync(ProcessStepResult processStepResultDto, ProcessCompletedEventDto processCompletedEventDto);
+    Task PublishCalculationResultForTotalGridAreaAsync(ProcessStepResult processStepResultDto, ProcessCompletedEventDto processCompletedEventDto);
+
+    Task PublishCalculationResultForEnergySupplierAsync(ProcessStepResult processStepResultDto, ProcessCompletedEventDto processCompletedEventDto, string energySupplierGln);
```

Comment:
Does this signature indicate too much responsibility? I would expect something more similar to:
```suggestion
    Task PublishAsync(IntegrationEvent integrationEvent);
```

Reply:
Could you elaborate on your approach? Perhaps create an example of how you would expect `IntegrationEvent` to look.

Reply:
A refactor of that size now would seem highly inefficient to me with the Outbox approach coming in a near feature.

Reply:
Okay 😄